AWSAuthSDK/Sources/AWSMobileClient/AWSMobileClient.swift

```diff
@@ -177,7 +177,7 @@ final public class AWSMobileClient: _AWSMobileClient {
     public func initialize(_ completionHandler: @escaping (UserState?, Error?) -> Void) {
         // Read awsconfiguration.json and set the credentials provider here
         initializationQueue.sync {
-            
+            self.keychain.migrateToCurrentAccessibility()
```

Comment:
Asking to confirm: it's not possible for the invocation to `migrateToCurrentAccesibility` to end up calling this `initialize` method again, right? In other words, no risk of a double `sync` on the same queue?